src/utils/__tests__/initilizerTests.js

```diff
@@ -0,0 +1,388 @@
+    plugin: 1,
+  });
+});
+
+test('init returns composed reducer given plugins', (assert) => {
+  const ctx = {
+    props: {
+      plugins: [
+        { reducer: { PLUGIN: () => ({ plugin: 0 }) } },
+        { reducer: { PLUGIN: () => ({ plugin: 1 }) } },
+      ],
+    },
+  };
+  const defaults = {
+    reducers: {
+      DEFAULTS: () => ({ defaults: true }),
+      PLUGIN: () => ({ plugin: false }),
+    },
+  };
+
+  const res = init.call(ctx, defaults);
+  assert.truthy(res);
+
+  assert.is(typeof res.reducers, 'function');
+  assert.deepEqual(Object.keys(res.reducers), ['DEFAULTS', 'PLUGIN']);
+  assert.deepEqual(res.reducers({}, { type: 'DEFAULTS' }), { defaults: true });
+  assert.deepEqual(res.reducers({}, { type: 'PLUGIN' }), { plugin: 1 });
+});
+
+test('init returns flattened/compacted reduxMiddleware given plugins', (assert) => {
+  const mw = _.range(0, 4).map(i => () => i);
+  const ctx = {
+    props: {
+      plugins: [
+        {},
+        { reduxMiddleware: [mw[0]] },
+        {},
+        { reduxMiddleware: [null, mw[1], undefined, mw[2], null] },
+        {},
+      ],
+      reduxMiddleware: [null, mw[3], undefined],
+    },
+  };
+  const defaults = {};
+
+  const res = init.call(ctx, defaults);
+  assert.truthy(res);
+
+  assert.deepEqual(res.reduxMiddleware, mw);
+});
+
+test('init sets context.components as expected given plugins', (assert) => {
+  const ctx = {
+    props: {
+      plugins: [
+        { components: { Plugin: 0, User: false } },
+        { components: { Plugin: 1 } },
+      ],
+      components: { User: true },
+    },
+  };
+  const defaults = {
+    components: {
+      Defaults: true,
+      Plugin: false,
+    },
+  };
+
+  const res = init.call(ctx, defaults);
+  assert.truthy(res);
+
+  assert.deepEqual(ctx.components, {
+    Defaults: true,
+    Plugin: 1,
+    User: true,
+  });
+});
+
+test('init sets context.settingsComponentObjects as expected given plugins', (assert) => {
+  const ctx = {
+    props: {
+      plugins: [
+        { settingsComponentObjects: { Plugin: 0, User: false } },
+        { settingsComponentObjects: { Plugin: 1 } },
+      ],
+      settingsComponentObjects: { User: true },
+    },
+  };
+  const defaults = {
+    settingsComponentObjects: {
+      Defaults: true,
+      Plugin: false,
+    },
+  };
+
+  const res = init.call(ctx, defaults);
+  assert.truthy(res);
+
+  assert.deepEqual(ctx.settingsComponentObjects, {
+    Defaults: true,
+    Plugin: 1,
+    User: true,
+  });
+});
+
+test('init sets context.events as expected given plugins', (assert) => {
+  const ctx = {
+    props: {
+      plugins: [
+        { events: { Plugin: 0, User: false } },
+        { events: { Plugin: 1 } },
+      ],
+      events: { User: true, User2: true },
+    },
+  };
+  const defaults = {};
+
+  const res = init.call(ctx, defaults);
+  assert.truthy(res);
+
+  assert.deepEqual(ctx.events, {
+    Plugin: 1,
+    User: false, // TODO: bug that plugins overwrite user events?
```

Comment:
@joellanciaux @ryanlanciaux is it intentional that `events` provided in `<Griddle events={...}>` are overwritten by `events` from plugins?